packages/core/react/src/ConnectionProvider.tsx

```diff
@@ -13,7 +13,16 @@ export const ConnectionProvider: FC<ConnectionProviderProps> = ({
     endpoint,
     config = { commitment: 'confirmed' },
 }) => {
-    const connection = useMemo(() => new Connection(endpoint, config), [endpoint, config]);
-
-    return <ConnectionContext.Provider value={{ connection }}>{children}</ConnectionContext.Provider>;
+    const [connection, setConnection] = useReducer(
+        (_: Connection, newConnection: Connection) => newConnection,
+        new Connection(endpoint, config)
+    );
```

Comment:
From https://reactjs.org/docs/hooks-reference.html#usereducer --

> useReducer is usually preferable to useState when you have complex state logic that involves multiple sub-values or when the next state depends on the previous one. useReducer also lets you optimize performance for components that trigger deep updates because you can pass dispatch down instead of callbacks.

What about this use case makes this a fit?